Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -771,53 +817,77 @@ namespace Terrain
+                AZStd::vector<NormalDataType> meshNormals;
 
                 {
                     SectorDataRequest request;
                     request.m_samplesX = GridVerts1D;
                     request.m_samplesY = GridVerts1D;
-                    request.m_worldStartPosition = AZ::Vector2(sector.m_worldX * gridMeters, sector.m_worldY * gridMeters);
+                    request.m_worldStartPosition = AZ::Vector2(sector->m_worldX * gridMeters, sector->m_worldY * gridMeters);
                     request.m_vertexSpacing = gridMeters / GridSize;
 
-                    AZStd::vector<HeightDataType> meshHeights;
-                    AZStd::vector<NormalDataType> meshNormals;
-                    GatherMeshData(request, meshHeights, meshNormals, sector.m_aabb);
-                    UpdateSectorBuffers(sector, meshHeights, meshNormals);
+                    GatherMeshData(request, meshHeights, meshNormals, sector->m_aabb, sector->m_hasData);
+                    if (sector->m_hasData)
+                    {
+                        UpdateSectorBuffers(*sector, meshHeights, meshNormals);
+                    }
                 }
 
-                if (m_config.m_clodEnabled)
+                if (m_config.m_clodEnabled && sector->m_hasData)
                 {
                     SectorDataRequest request;
                     uint16_t gridSizeNextLod = (GridSize >> 1);
                     request.m_samplesX = gridSizeNextLod + 1;
                     request.m_samplesY = gridSizeNextLod + 1;
-                    request.m_worldStartPosition = AZ::Vector2(sector.m_worldX * gridMeters, sector.m_worldY * gridMeters);
+                    request.m_worldStartPosition = AZ::Vector2(sector->m_worldX * gridMeters, sector->m_worldY * gridMeters);
                     request.m_vertexSpacing = gridMeters / gridSizeNextLod;
 
-                    AZ::Aabb dummyAabb = AZ::Aabb::CreateNull(); // don't update the sector aabb based on only the clod vertices.
+                    AZ::Aabb dummyAabb = AZ::Aabb::CreateNull(); // Don't update the sector aabb based on only the clod vertices.
+                    bool terrainExists = false;
                     AZStd::vector<HeightDataType> meshLodHeights;
                     AZStd::vector<NormalDataType> meshLodNormals;
-                    GatherMeshData(request, meshLodHeights, meshLodNormals, dummyAabb);
-                    UpdateSectorLodBuffers(sector, meshLodHeights, meshLodNormals);
+                    GatherMeshData(request, meshLodHeights, meshLodNormals, dummyAabb, terrainExists);
+                    if (!terrainExists)
+                    {
+                        // It's unlikely but possible for the higher lod to have data and the lower lod to not. In that case 
+                        // meshLodHeights will be empty, so fill it with values that represent "no data".
+                        AZStd::fill(meshLodHeights.begin(), meshLodHeights.end(), uint16_t(0xFFFF));
```

Comment:
This would be a great place to reuse that named constant... :)

Reply:
😛